### ISSUE TITLE
setIndentation causes accessibility flag when child lists are not wrapped in li element

### ISSUE THREAD

**Issue by sayers24** [microsoft/roosterjs#1152]
**Describe the bug**

**To Reproduce**
1. Go to the demo site - https://microsoft.github.io/roosterjs/
2. Create a list
3. Indent one of the list items
4. Run accessibility insights browser plugin - fast pass 

**Expected behavior**
The structure for nested lists should be as follows
```
<ul>
  <li>List item</li>
  <li>
    <ul>
      <li>Sub list item</li>
    </ul>
  </li>
</ul>
```

**Screenshots**
![image](https://user-images.githubusercontent.com/2318860/182414492-c5d3e489-f410-4de5-bd24-6f9574441e10.png)

**Comment by JiuqingSong** [comment]
With the expected HTML code, the list will be rendered as below (both Chrome and Firefox), which is incorrect.
![image](https://user-images.githubusercontent.com/23065085/182908548-a6166b2c-ff89-454a-9ef7-a79c078f7827.png)

**Comment by romanisa** [comment]
@sayers24 Scott, can you please help explain what is the actual accessibility issue leading to this?

**Comment by sayers24** [comment]
@romanisa, sure. The accessibility issue is that the setIndentation is creating a `<ul>` element as a child of the parent `<ul>`. According to the w3c standards, `ul` cannot be a valid child of `ul`. Valid children for `<ul>` and `<ol>` are the `<li>` elements. It is suggested by [docs](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/HTML_text_fundamentals#nesting_lists) to wrap nested lists within an `<li>` element.

@JiuqingSong - The double bullets could be avoided with css as follows:
```
<ul>
  <li>List item</li>
  <li style="list-style-type: none;">
    <ul>
      <li>Sub list item</li>
    </ul>
  </li>
</ul>
```

**Comment by JiuqingSong** [comment]
You can write a plugin in your project to always format the list in the expected way. It can handle the following event to do this format change:
- ContentChanged event: happens when user indent/outdent a list item, change the list structure that is just changed
- BeforePaste event: happens when content is about to be pasted, see if there is list inside and format it
- EditorReady event: happens when editor is initialized, for handling those existing content that is loaded directly into editor

**Comment by sayers24** [comment]
I could do this to address the accessibility compliance issue for my team's implementation but figured it would be better to address accessibility issues at the source so everyone could benefit.

**Comment by JiuqingSong** [comment]
It is always welcome to contribute back with your new plugin once it is ready.

But I actually still have a question, what is the real impact to end user from this issue? Does it cause any behavior issue, or announcer issue? If it just cause some failure in an accessibility tool but no real customer impact, I don't feel we should take it for now.

**Comment by jvillalobos** [comment]
Closing until we get more info.
